controllers/cloud.redhat.com/providers/autoscaler/simple.go

```diff
@@ -0,0 +1,148 @@
+package autoscaler
+
+import (
+	"fmt"
+	"log"
+
+	res "k8s.io/apimachinery/pkg/api/resource"
+
+	crd "github.com/RedHatInsights/clowder/apis/cloud.redhat.com/v1alpha1"
+	"github.com/RedHatInsights/clowder/controllers/cloud.redhat.com/config"
+	"github.com/RedHatInsights/clowder/controllers/cloud.redhat.com/providers"
+	deployProvider "github.com/RedHatInsights/clowder/controllers/cloud.redhat.com/providers/deployment"
+	apps "k8s.io/api/apps/v1"
+	v2 "k8s.io/api/autoscaling/v2"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func ProvideSimpleAutoScaler(app *crd.ClowdApp, appConfig *config.AppConfig, sp *providers.Provider, deployment crd.Deployment) error {
+	cachedDeployment, err := getDeploymentFromCache(&deployment, app, sp)
+	if err != nil {
+		log.Println(err)
+		return err
+	}
+	deploymentHPA := makeDeployemntSimpleHPA(&deployment, app, appConfig, cachedDeployment)
+	hpaResource := deploymentHPA.getResource()
+
+	err = sp.Client.Create(sp.Ctx, &hpaResource)
+	if err != nil {
+		fmt.Println("HPA Error: ", err)
+	}
+	return nil
+}
+
+//Get the core apps.Deployment from the provider cache
+//This is in AutoScalerSimpleProvider because we need access to the provider cache
+func getDeploymentFromCache(clowdDeployment *crd.Deployment, app *crd.ClowdApp, sp *providers.Provider) (*apps.Deployment, error) {
+	nn := app.GetDeploymentNamespacedName(clowdDeployment)
+	d := &apps.Deployment{}
+	if err := sp.Cache.Get(deployProvider.CoreDeployment, d, nn); err != nil {
+		return d, err
+	}
+	return d, nil
+}
+
+func makeDeployemntSimpleHPA(deployment *crd.Deployment, app *crd.ClowdApp, appConfig *config.AppConfig, coreDeployment *apps.Deployment) deployemntSimpleHPA {
+	return deployemntSimpleHPA{
+		deployment:     deployment,
+		app:            app,
+		appConfig:      appConfig,
+		coreDeployment: coreDeployment,
+	}
+}
+
+type deployemntSimpleHPA struct {
+	deployment     *crd.Deployment
+	app            *crd.ClowdApp
+	appConfig      *config.AppConfig
+	coreDeployment *apps.Deployment
+}
+
+func (d *deployemntSimpleHPA) getResource() v2.HorizontalPodAutoscaler {
```

Comment:
We should probably be passing pointers to these structs, and not creating them each time.